public/apps/configuration/panels/audit-logging/audit-logging.tsx

```diff
@@ -30,22 +30,30 @@ import {
   renderComplianceSettings,
   renderGeneralSettings,
   renderStatusPanel,
+  updateAuditLogging,
 } from '../../utils/audit-logging-view-utils';
+import { AuditLoggingSettings } from './types';
 
 export function AuditLogging(props: AppDependencies) {
-  // TODO: modify the logic when making the post calls for update.
-  const [auditLoggingEnabled, setAuditLoggingEnabled] = useState<boolean>(false);
-  const [configuration, setConfiguration] = useState<any>({});
-  const onSwitchChange = () => {
-    setAuditLoggingEnabled((prevAuditLoggingEnabled) => !prevAuditLoggingEnabled);
+  const [configuration, setConfiguration] = useState<AuditLoggingSettings>({});
+
+  const onSwitchChange = async () => {
+    try {
+      const updatedConfiguration = Object.assign({}, configuration);
```

Comment:
I think we can use `const updatedConfiguration = {...configuration};` here

Reply:
Sure. Looks more elegant